pkg/translator/loki/logs_to_loki_test.go

```diff
@@ -18,11 +18,295 @@ import (
+		name     string
+		logs     plog.Logs
+		expected map[string]PushRequest
+	}{
+		{
+			name: "tenant from logs attributes",
+			logs: func() plog.Logs {
+				logs := plog.NewLogs()
+				rl := logs.ResourceLogs().AppendEmpty()
+
+				sl := rl.ScopeLogs().AppendEmpty()
+				logRecord := sl.LogRecords().AppendEmpty()
+				logRecord.Attributes().PutStr(hintTenant, "tenant.id")
+				logRecord.Attributes().PutStr("tenant.id", "1")
+				logRecord.Attributes().PutInt("http.status", 200)
+
+				sl = rl.ScopeLogs().AppendEmpty()
+				logRecord = sl.LogRecords().AppendEmpty()
+				logRecord.Attributes().PutStr(hintTenant, "tenant.id")
+				logRecord.Attributes().PutStr("tenant.id", "2")
+				logRecord.Attributes().PutInt("http.status", 200)
+
+				return logs
+			}(),
+			expected: map[string]PushRequest{
+				"1": {
+					PushRequest: &logproto.PushRequest{
+						Streams: []logproto.Stream{
+							{
+								Labels: `{exporter="OTLP", tenant.id="1"}`,
+								Entries: []logproto.Entry{
+									{
+										Line: `{"attributes":{"http.status":200}}`,
+									},
+								}},
+						},
+					},
+				},
+				"2": {
+					PushRequest: &logproto.PushRequest{
+						Streams: []logproto.Stream{
+							{
+								Labels: `{exporter="OTLP", tenant.id="2"}`,
+								Entries: []logproto.Entry{
+									{
+										Line: `{"attributes":{"http.status":200}}`,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "tenant from resource attributes",
+			logs: func() plog.Logs {
+				logs := plog.NewLogs()
+				rl := logs.ResourceLogs().AppendEmpty()
+				rl.Resource().Attributes().PutStr(hintTenant, "tenant.id")
+				rl.Resource().Attributes().PutStr("tenant.id", "11")
+
+				sl := rl.ScopeLogs().AppendEmpty()
+				logRecord := sl.LogRecords().AppendEmpty()
+				logRecord.Attributes().PutInt("http.status", 200)
+
+				rl = logs.ResourceLogs().AppendEmpty()
+				rl.Resource().Attributes().PutStr(hintTenant, "tenant.id")
+				rl.Resource().Attributes().PutStr("tenant.id", "12")
+
+				sl = rl.ScopeLogs().AppendEmpty()
+				logRecord = sl.LogRecords().AppendEmpty()
+				logRecord.Attributes().PutInt("http.status", 200)
+
+				return logs
+			}(),
+			expected: map[string]PushRequest{
+				"11": {
+					PushRequest: &logproto.PushRequest{
+						Streams: []logproto.Stream{
+							{
+								Labels: `{exporter="OTLP", tenant.id="11"}`,
+								Entries: []logproto.Entry{
+									{
+										Line: `{"attributes":{"http.status":200}}`,
+									},
+								}},
+						},
+					},
+				},
+				"12": {
+					PushRequest: &logproto.PushRequest{
+						Streams: []logproto.Stream{
+							{
+								Labels: `{exporter="OTLP", tenant.id="12"}`,
+								Entries: []logproto.Entry{
+									{
+										Line: `{"attributes":{"http.status":200}}`,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "use tenant resource attributes if both logs and resource attributes provided",
```

Comment:
Is this documented? If not, could you please mention it somewhere?

Reply:
add a sentence in the method doc.